### ISSUE TITLE
Gitbook Editor reverts any changes made externally

### ISSUE THREAD

**Issue by Deozaan** [GitbookIO/feedback#186]
### What is the current behavior?

Make a change to a file using a text editor of your choice. Save it. Now open that file in the Gitbook Editor. Your file has been reverted without any notification or confirmation.

### What is the expected behavior?

It should detect the file as changed/unsaved, or at the very least just ask if I want to keep the change or revert to the most recently committed version.

#### For GitBook Editor

- Operating System: Windows 10 Pro x64
- (Desktop) Editor version: 6.2.1
- Was the bug present in previous versions: Not sure as I never tried before.

**Comment by SamyPesse** [comment]
It will be fixed by https://github.com/GitbookIO/feedback/issues/101

The editor only works with files in the GIT history.

**Comment by Deozaan** [comment]
@SamyPesse I'm not sure #101 is related. As I understand it, #101 is about creating new files from outside of the GitBook Editor.

These files I'm talking about in this issue were already in the git history, having been created from within the GitBook Editor already. In fact, I had added and modified them many times before using the GitBook Editor. But when I edited (not created) the files externally to the Editor, they became reverted when I opened the Editor again.

**Comment by SamyPesse** [comment]
Yes, technically this is the same issue, #101 only describe one aspect of the problem. But basically this is about supporting edition/creation outside the editor.

**Comment by SamyPesse** [comment]
We just released the first beta of 6.3.0, you can download it and read the full changelog at [gitbook.com/editor/releases/all](https://www.gitbook.com/editor/releases/all).

it adds basic synchronization with local files and git repository. Can you test and let me know if it works as expected?

**Comment by Deozaan** [comment]
I opened up GitBook Editor to a page, then opened that page in Notepad++ and added a line, then saved it. Almost immediately I saw a popup dialog in GitBook Editor telling me the file had changed externally and asked me to select which file(s) I wanted to sync changes with, and which files' changes I wanted to discard.

I selected the file I edited (actually it was already selected) and pressed sync. Within seconds I saw the icon in the corner show that there was 1 local change that hadn't been synced with the GitBook repository, and I saw the source in the editor update to show the change. But the preview window didn't update for several seconds, and then GitBook Editor locked up, using about 20% CPU (of my 8-cores) for so long that I was halfway finished writing this response (up to this point) before the application finally started responding again.

But I had pressed the X button in the corner to close it while it was still locked up, so it closed immediately when it started responding again.

So I just tried it again, this time undoing the little test edit I made, and I'm seeing similar behavior. It asked me what to sync, I press the sync button, the little cloud icon updates to show 2 unsynced changes now, the source window updated, a few seconds later the preview window updated, and the entire program is unresponsive and using ~20% CPU usage for about maybe 2-3 minutes before finally becoming responsive again and after that the CPU usage gradually goes down to normal levels.
